docs/source/daml/intro/6_Parties.rst

```diff
@@ -0,0 +1,197 @@
+
+There's a new problem here: There is no way for Alice to issue or transfer this ``Iou`` to Bob. To get an ``Iou`` with Bob's signature as ``owner`` onto the ledger, his authority is needed.
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- IOU_SCENARIO_BEGIN
+  :end-before: -- IOU_SCENARIO_END
+
+This may seem awkward, but notice that the ``ensure`` clause is gone from the ``Iou`` again. The above ``Iou`` can contain negative values so Bob should be glad that ``Alice`` cannot put his signature on any ``Iou``.
+
+You'll now learn a couple of common of ways of buildng issuance and transfer workflows for the above ``Iou``, before diving into the authorization model in full.
+
+Use propose-accept workflows for one-off authorizaiton
+------------------------------------------------------
+
+If there is no standing relationship between Alice and Bob, Alice can propose the issuance of an Iou to Bob, giving him the choice to accept. You can do so by introducing a proposal contract ``IouProposal``:
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- IOU_PROPOSAL_BEGIN
+  :end-before: -- IOU_PROPOSAL_END
+
+Note how we have used the fact that templates are records here to store the ``Iou`` in a single field.
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- IOU_PROPOSAL_SCENARIO_BEGIN
+  :end-before: -- IOU_PROPOSAL_SCENARIO_END
+
+The ``IouProposal`` contract carries the authorithy of ``iou.issuer`` by virtue of them being a signatory. By exercising the ``IouProposal_Accept`` choice, Bob adds his authority to that of Alice, which is why an ``Iou`` with both signatories can be created in the context of that choice.
+
+The choice is called ``IouProposal_Accept``, not ``Accept``, because propose-accept patterns are very common. In fact, you'll see another one just below. As each choice defines a record type, you cannot have two choices of the same name in scope. It's a good idea to qualify choice names to ensure uniqueness.
+
+The above solves issuance, but not transfers. You can solve transfers exactly the same way, though, by creating a ``TransferProposal``:
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- TRANSFER_PROPOSAL_BEGIN
+  :end-before: -- TRANSFER_PROPOSAL_END
+
+In addition to defining the signatories of a contract, ``signatory`` can also be used to extract the signatories from another contract. Instead of writing ``signatory (signatory iou)``, you could write ``signatory iou.issuer, iou.owner``.
+
+Note also how ``newOwner`` is given multiple choices using a single ``controller newOwner can`` block. The ``IouProposal`` had a single signatory so it could be cancelled easily by archiving it. Without a ``Cancel `` choice, the ``newOwner`` could abuse an open TransferProposal as an option. The triple ``Accept``, ``Reject``, ``Cancel`` is common to most proposal templates.
```

Comment:
```suggestion
Note also how ``newOwner`` is given multiple choices using a single ``controller newOwner can`` block. The ``IouProposal`` had a single signatory so it could be cancelled easily by archiving it. Without a ``Cancel`` choice, the ``newOwner`` could abuse an open TransferProposal as an option. The triple ``Accept``, ``Reject``, ``Cancel`` is common to most proposal templates.
```